File: 0021-merge-two-sorted-lists/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode temp = list1;
        ListNode temp1 = list2;
        ListNode curr = new ListNode();
        ListNode ans=curr;
        
        while(temp != null && temp1 != null){
            if(temp.val < temp1.val){
                ans.next=temp;
                temp = temp.next;
            }
            else{
                ans.next = temp1;
                temp1 =temp1.next;
            }
            ans = ans.next;
        }
        if(temp==null){
            ans.next=temp1;
        }else if(temp1==null){
            ans.next=temp;
        }
        return curr.next;
    }
}

File: 0021-merge-two-sorted-lists/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* mergeTwoLists(struct ListNode* list1, struct ListNode* list2) {
    struct ListNode* dummy=(struct ListNode *)malloc(sizeof(struct ListNode));
    struct ListNode* cur=dummy;
    while(list1!=NULL && list2!=NULL){
        if(list1->val < list2->val){
            cur->next=list1;
            list1=list1->next;
        }else{
            cur->next=list2;
            list2=list2->next;
        }
        cur=cur->next;
    } 
    if(list1==NULL){
        cur->next=list2;
    }else{
        cur->next=list1;
    }
    return dummy->next;
}
